### ISSUE TITLE
Order of contributors in project metadata is not preserved

### ISSUE THREAD

**Issue by hannes-ucsc** [DataBiosphere/azul#4364]
https://humancellatlas.slack.com/archives/C9XD6L0AD/p1658915484833189

**Comment by melainalegaspi** [comment]
@hannes-ucsc :"spike to prototype a solution in the form of a PR followed by a WIP review. The biggest unknown is whether ElasticSearch maintains order of array elements in the document source."

**Comment by achave11** [comment]
@hannes-ucsc and I confirmed that ElasticSearch maintains order of array elements by inferring Azul source code and DataBrowser responses.

@hannes-ucsc: "We should invest in an ordered set class. We looked at ordered-set and orderedset from pypi. The former has a disconcerting bug in .pop() and it uses a backing list in addition to a dict which seems superfluous given that dicts maintain insertion order nowadays. The latter is unmaintained and doesn't install on python 3.9+. Write a minimal implementation of ordered set that just uses a backing dictionary. Add that implementation to metadata-api, use it there to store contributors, and other properties currently typed as set. Also use it in Azul to store contributors, institutions, etc."
